### PR TITLE
[AUTH-1562] Add a tool to continuously generate login/logout requests to the idp

### PR DESCRIPTION
**Pull request by tomthorogood**
Preview:

Screen Shot 2022-02-03 at 2 31 46 PM

Example run (running 5 attempts w/ 5 seconds each): https://github.com/UWIT-IAM/uw-idp-web-tests/actions/runs/1792108277

Note that because I renamed the workflow file, the workflow listed in the Actions index (and in the doc herein) will not work until this PR is merged.

### CODE REVIEW

**Comment by tomthorogood** [comment]
Ignore that failing check, it was spurious.

File: tests/conftest.py [hunk 1]
```diff
@@ -16,6 +16,12 @@ def pytest_addoption(parser):
                      help="Use this if you want to provide a settings YAML file yourself.")
     group.addoption('--settings-profile', default='base',
                      help="Use this if you want to use settings from a specific root node of the settings file.")
+    group.addoption('--lb-num-loops', default=0,
+                    help="If set to a number greater than 0, "
+                         "the test_generate_requests.py test will make IdP requests the given number of times.")
+    group.addoption('--lb-sleep-time', default=60,
+                    help="The number of seconds to sleep between login attempts when running the 'load-balancer' "
+                         "mark.")
```
Comment (JimTomlinson-UW) [inline tests/conftest.py]: I'm a python n00b; what does the word 'mark' refer to? Is this a pytest reference?

**Review by JimTomlinson-UW — APPROVED** [review]

**Comment by tomthorogood** [comment]
@JimTomlinson-UW (for some reason I can't respond to your comment) -- It is a pytest reference, but this comment should be removed, because I ditched the idea of a load-balancer mark, so this documentation is out of date. :)

https://docs.pytest.org/en/6.2.x/mark.html

Marking tests allows you to filter tests based on their mark. For instance:

```
@pytest.mark('slow')
def test_slow_thing():
    time.sleep(1000)

def test_fast_thing():
    time.sleep(0.25)
```

```
pytest  # Runs all tests, even the slow one
pytest -k 'not slow'  # Skips the slow test
pytest -m 'slow'  # Only runs the slow test
```

**Comment by mar235av** [comment]
Looks good to me. I was going to approve but Tom has already completed the merge.

**Pull request merged**